reader/tests/system/test_client.py

```diff
@@ -33,3 +41,113 @@ def test_invalid_requests(json_client):
     for request in requests:
         response = json_client.post(Route.GET.URL, request)
         assert_error_response(response, ERROR_CODES.INVALID_REQUEST)
+
+
+def test_concurrent_requests():
+    os.environ["DATASTORE_MAX_CONNECTIONS"] = "2"
+    # TODO: clear cache of env service in app
```

Comment:
@FinnStutzenstein I couldn't figure out how to do this; currently, the max connections variable is just hardcoded to 2 in the `PgConnectionHandlerService`. The `EnvironmentService` of the newly created app is not accessible from this scope, so how do I clear the cache of it? Why do we even cache env variables, is that much of a performance gain?

Reply:
>  Why do we even cache env variables

IDK, you could remove this, but it won't solve the issue.

I wonder, why you do not use `https://github.com/OpenSlides/openslides-datastore-service/blob/master/reader/reader/app.py`. The DI is not set up in your code, right? I do not think it should work in the way you are doing this.

If you call `register_services`, then you can aquire the env service via `injector.get` before creating the application.

Reply:
The problem when doing this is that somehow, another `EnvironmentService` is grabbed than the one used by the client. In fact, when logging it, there were 3 different ones created. If I called `register_services` and removed the caching, however, the test passed. Very strange. Well, my solution now is cleaner and just patches the `maxconn` of the pool and semaphore. I previously thought this wasn't possible without additional effort.

Reply:
BTW, it's even stranger since the injected `ReadDatabase` is grabbed and patched correctly...